lib/std/os/windows.zig

```diff
@@ -2859,12 +2859,12 @@ pub const LONG = i32;
 pub const ULONG64 = u64;
 pub const ULONGLONG = u64;
 pub const LONGLONG = i64;
-pub const HLOCAL = HANDLE;
-pub const LANGID = c_ushort;
+pub const HLOCAL = *anyopaque;
```

Comment:
this change in not justified in your commit descriptions

Reply:
I think `*opaque{}` makes the most sense here and matches what win32metadata is currently doing (zigwin32 is out-of-date and this has changed since).  win32metadta has defined its own Handle type for `LocalAlloc`.  The result of `LocalAlloc` isn't meant to be used anywhere `HANDLE` is meant to be used as far as I know.

That being said, another alternative is to remove `HLOCAL` and `LocalFree`, doesn't seem like zig is using them.

P.S. I see you updated your PR description but I believe Andrew was wanting you to put your comments in the commit description (people's clones of the git don't have access to github PR descriptions).

Reply:
@expikr FWIW, to be "more-proper", we should have a way to access/twiddle the lower 2 bits.
In a HANDLE and other such kernel handles, the lowest 2 bits are ignored and can be used by the user/program to "tag" various handles going in or out of the kernel.
So a more correct type might be:
```zig
packed struct {
  bits: u2,
  opaque_bits: std.meta.Int(.signed, @bitSizeOf(usize)-2),
}
```
or similar?

Reply:
> but Windows have other objects that have `H****` handle types:
> 
> > Not very well known is that the bottom two bits of kernel HANDLEs are always zero; in other words, their numeric value is always a multiple of 4. Note that this applies only to kernel HANDLEs; it does not apply to pseudo-handles or to any other type of handle (USER handles, GDI handles, multimedia handles…) Kernel handles are things you can pass to[ the CloseHandle function](http://msdn.microsoft.com/library/en-us/sysinfo/base/closehandle.asp).
> > — https://devblogs.microsoft.com/oldnewthing/20050121-00/?p=36633

Yes indeed e.g. GDI handles exist.
Perhaps we should have something like:
```zig
pub const HANDLE = extern union {
    pub const KERNEL = packed struct {
      bits: u2,
      opaque_bits: std.meta.Int(.signed, @bitSizeOf(usize)-2),
    };
    kernel: KERNEL,
    user: *anyopaque,
    gdi: *anyopaque,
}
```

And then e.g. the `CloseHandle ` function mentioned would take a `HANDLE.KERNEL` instead of just a HANDLE?

Reply:
I don't have enough practical experience with `HLOCAL` to have a worthwhile opinion here.

One data point that might be relevant, though, is how these handles are defined with [`STRICT`](https://learn.microsoft.com/en-us/windows/win32/winprog/strict-type-checking) enabled. From my reading of `windef.h`, `HLOCAL` bypasses `STRICT` and is always the same type as `HANDLE` (i.e. it is defined as `typedef HANDLE HLOCAL;` instead of using the `DECLARE_HANDLE` macro which gives a handle a distinct type when `STRICT` is enabled).

Reply:
> makein HCURSOR a synonym of HICON

This is one I think I'd actually recommend against unless there's a particular use case in mind. You're right that `WinDef.h` defines them as aliases, and they may be able to be used interchangeably in certain circumstances, but given that there are things like separate [DestroyIcon](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-destroyicon) and [DestroyCursor](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-destroycursor) functions, enforcing type safety seems like a good idea.